test/unit/sources/hentai_foundry_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+
+module Sources
+  class HentaiFoundryTest < ActiveSupport::TestCase
+    context "The source for a hentai foundry picture" do
+      setup do
+        @image_1 = Sources::Strategies.find("https://www.hentai-foundry.com/pictures/user/Afrobull/795025/kuroeda")
+        @image_2 = Sources::Strategies.find("https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png")
+      end
+
+      should "get the illustration id" do
+        assert_equal("795025", @image_1.illust_id)
+        assert_equal("795025", @image_2.illust_id)
+      end
+
+      should "get the artist name" do
+        assert_equal("Afrobull", @image_1.artist_name)
+        assert_equal("Afrobull", @image_2.artist_name)
+      end
+
+      should "get the artist commentary title" do
+        assert_equal("kuroeda", @image_1.artist_commentary_title)
+        assert_equal("kuroeda", @image_2.artist_commentary_title)
+      end
+
+      should "get profile url" do
+        assert_equal("https://www.hentai-foundry.com/user/Afrobull", @image_1.profile_url)
+        assert_equal("https://www.hentai-foundry.com/user/Afrobull", @image_2.profile_url)
+      end
+
+      should "get the image url" do
+        assert_equal("https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png", @image_1.image_url)
+        assert_equal("https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png", @image_2.image_url)
+      end
+
+      should "get the canonical url" do
+        assert_equal("https://www.hentai-foundry.com/pictures/user/Afrobull/795025", @image_1.canonical_url)
```

Comment:
should the `page_url` be the canonical source?
I thought the `image_url` was prefered if it was possible to reconstruct the `page_url` from it?

Fwiw: for [post #3858510](https://danbooru.donmai.us/posts/3858510) (NSFW), the `source` is what would be the `image_url`

Reply:
You're right, I'll change it.